code/game/objects/items/gamemode/survivor.dm

```diff
@@ -0,0 +1,354 @@
+        return
+
+    if(!ishuman(user)) 
+        return
+
+    if(!user.mind)
+        to_chat(user, "<span class='warning'>It doesn't seem to do anything for you.</span>")
+        return
+
+    var/area/A = get_area(user)
+    if(istype(A) && A.ceiling >= CEILING_METAL)
+        to_chat(user, "<span class='warning'>You have to be outside or under a glass ceiling to activate this.</span>")
+        return
+
+    user.visible_message("<span class='notice'>[user] starts setting up [src] on the ground.</span>",
+    "<span class='notice'>You start setting up [src] on the ground and inputting all the data it needs.</span>")
+
+    if(!do_after(user, activation_time, TRUE, 5, BUSY_ICON_FRIENDLY,, TRUE))
+        return
+
+    log_game("[key_name(user)] placed \the [src] near [AREACOORD(A.loc)]")
+
+    user.transferItemToLoc(src, user.loc)
+    anchored = TRUE
+    w_class = 10
+    update_icon()
+        
+
+
+/obj/item/laptop/rescue/proc/reset_state()
+    icon_state = "tandy0"
+    w_class = initial(w_class)
+    anchored = FALSE
+    SetLuminosity(initial(luminosity))
+    update_icon()
+
+
+/obj/item/laptop/rescue/attack_alien(mob/living/carbon/Xenomorph/M)
+    if(!anchored)
+        to_chat(M, "<span class='warning'>It's already broken.</span>")
+        return
+
+    M.animation_attack_on(src)
+    M.visible_message("<span class='danger'>\The [M] slices [src] apart!</span>", \
+        "<span class='danger'>You slice [src] apart!</span>", null, 5)
+    playsound(loc, "alien_claw_metal", 25, 1)
+
+    if (!anchored)
+        return
+
+    reset_state()
+   
+    if(prob(10))
+        new /obj/effect/decal/cleanable/blood/oil(loc)
+
+/obj/item/beacon/rescue
+    name = "Rescue beacon"
+    desc = "A rugged, glorified laser pointer capable of sending a beam into space. Activate and throw this to call for someone to rescue you."
+    icon_state = "motion0"
+    resistance_flags = UNACIDABLE|INDESTRUCTIBLE
+    w_class = WEIGHT_CLASS_BULKY
+    luminosity = 4 
+
+    var/activated = FALSE
+    var/activation_time = 10 SECONDS
+    var/icon_activated = "motion2"
+
+    var/distress_timer = 10 MINUTES
+    var/required_components = list(
+        /obj/item/cell,
+    )
+
+    // Requirements for nearby components that need setting up
+    var/scan_range = 2
+    var/required_nearby = list(
+        /obj/item/laptop/rescue,
+    )
+
+    // Contents list, for when it gets broken. We spit everything out again
+    var/internal_components = list()
+
+    // Custom health, It can't be destroyed but it can be stopped by reducing health to 0
+    max_integrity = 100
+    obj_integrity = 0
+
+    // Timer to remove the distress beacon if we get distroyed
+    var/noise_timer_id = null
+    var/beacon_timer_id = null
+
+
+/obj/item/beacon/rescue/examine()
+    . = ..() // show parent examines (if any) first
+    if (beacon_timer_id)
+        to_chat(usr, "<span class='notice'>It shows [timeleft(beacon_timer_id) / 10] seconds left.</span>")
```

Comment:
Examine has a parameter, you should define it and use it (user) rather than usr.